### ISSUE TITLE
Splitting the core.phel

### ISSUE THREAD

**Issue by jenshaase** [jenshaase/phel-lang#169]
Make it possible to split the core.phel file in multiple file that have the same namespace.

**Comment by Chemaclass** [comment]
@jenshaase but do we want them to be in the same namespace? 🤔 
Think about having N files (each file with its different group of functions). Each one of them will share some common responsibilities. Isn't true that by having different namespaces it won't load the ones that we don't really want in our programs and therefore will it be faster to execute?
Same principle as we currently have for `http` and `html` phel's files.

What I am saying is that it might be OK to have some core functions in the core, and they will be loaded out of the box (similar to the `Util` package in Java). But if you want to use something different from another package then you have to import/require it. 

Doing this approach the only thing that we need to care of is how do we actually want to split the core, we don't need to alter any current logic. What do you think?

**Comment by jenshaase** [comment]
Ok, but then the question is when to split stuff and when not.

For example, do you want to put the `+` operation in a `math` namespace and import it everytime you need it?

**Comment by Chemaclass** [comment]
This is where the complexity of this task is about. I personally think the basics arithmetic, casting & comparation should be in the `core.phel`, but yeah I don't know now... maybe it's ok to split the "arithmetic's" in their own module as well... 
I am trying to figure out how would be the perfect scenario, but I don't have the answer now. First, we need to visualize how do we want to have it as a goal, to do that I am looking at how Clojure have grouped their functions, something like https://clojure.org/api/cheatsheet or https://clojuredocs.org/clojure.core

**Comment by JesusValera** [comment]
I consider the basic operations should be in the `core`.
Something similar as any programming language does, you don't need to import a package to `sum` two numbers.
The opposite with, for example, the `pi` number, in this scenario, you would really need to import the `math` then.

**Comment by jenshaase** [comment]
First thing I did now is to split the core lib into multiple files and load them in the `core.phel` file. Therefore, I added the `load` function.

@Chemaclass Regarding your proposal:
* Splitting the set operation would be ok for me.
* Sequence operations and functions like `map`, `reduce` and `filter` are essential for functional programming. You will use them nearly very where. Therefore, I don't think it is good idea to split them.
* Print operations can be split but if you like to debugging you code with print statements you always have to import the print operations. After that you probably will forget to remove them from the imports.
